### PR TITLE
Automate builds with GitHub Actions

### LINKED ISSUE #26: Create action that builds and deploys Nikola site to pages
https://getnikola.com/handbook.html#deploying-to-github

### PR DESCRIPTION
**Pull request by maxamillion**
Fixes #26

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+jobs:
+  nikola_build:
+    runs-on: ubuntu-latest
+    name: 'Deploy Nikola to GitHub Pages'
+    steps:
+    - name: Check out
+      uses: actions/checkout@v2
+    - name: Build and Deploy Nikola
+      uses: getnikola/nikola-action@v2
```
Comment (oraNod) [inline .github/workflows/main.yml:11]:
```suggestion
      uses: getnikola/nikola-action@v8
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+on: [push]
+
+jobs:
+  nikola_build:
+    runs-on: ubuntu-latest
+    name: 'Deploy Nikola to GitHub Pages'
+    steps:
+    - name: Check out
+      uses: actions/checkout@v2
```
Comment (oraNod) [inline .github/workflows/main.yml:9]:
```suggestion
      uses: actions/checkout@v3
```

**Review by oraNod — COMMENTED** [review]
From a bit of testing it looks like older versions of the action fail with `detected dubious ownership in repository at '/github/workspace'`

**Comment by oraNod** [comment]
@maxamillion Should we initialize the site with a basic "hello world" that uses the default theme?

**Comment by maxamillion** [comment]
@oraNod probably, yes ... want me to get a different PR in for that first?

**Comment by oraNod** [comment]
@maxamillion Yes, please, if you don't mind.

**Comment by oraNod** [comment]
I also just pushed a gitignore file that I think we need for the action, if that's OK.

**Comment by oraNod** [comment]
@maxamillion Hey I hope you don't mind but I've pushed a couple of changes as I've been testing this out a little on my fork.

I wonder about the deployment branch name. Should we just go with `gh-pages` instead of `prod` to avoid ambiguity? https://github.com/ansible-community/community-website/blob/ad0cc1698ae868a0fdd3e4906b11096dc22c843d/conf.py#L688

I'm not sure whether this will actually be hosted on pages or not so maybe the "prod" branch doesn't technically make sense?

@cybette Do you have any opinions?

Thanks!

**Comment by maxamillion** [comment]
> @maxamillion Hey I hope you don't mind but I've pushed a couple of changes as I've been testing this out a little on my fork.
> 
> I wonder about the deployment branch name. Should we just go with `gh-pages` instead of `prod` to avoid ambiguity? https://github.com/ansible-community/community-website/blob/ad0cc1698ae868a0fdd3e4906b11096dc22c843d/conf.py#L688
> 
> I'm not sure whether this will actually be hosted on pages or not so maybe the "prod" branch doesn't technically make sense?
> 
> @cybette Do you have any opinions?
> 
> Thanks!

Sounds good to me! I just picked a name so there was one. I have no preference. 🙂

**Review by oraNod — APPROVED** [review]

**Review by Spredzy — APPROVED** [review]
Looks good to me based on https://getnikola.com/handbook.html#deploying-to-github.

> GITHUB_DEPLOY_BRANCH is the branch where Nikola-generated HTML files will be deployed. It should be gh-pages for project pages and master for user pages (user.github.io).

Wondering if the branch needs to be created first of if it will be automatically crreated.

**Comment by cybette** [comment]
oops I assumed wrong :sweat_smile:  looks like we need the branch manually created after all

**Pull request merged**